### DISCUSSION TITLE
How to know which grain corresponds to the element orientation output by FEPX so as to obtain KAM

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/neper#563] [Q&A] [answered]
I plan to get KAM through post-processing，where The KAM is defined as the average misorientation between a kernel simulation point and its surrounding points excluding those out of the considered grain.

But fepx outputs the element orientation or the average grain orientation through post-processing of neper, but this is not enough. I need to know which elements are in the same grain in order to obtain KAM through my calculation formula. So what should i do ？

**Comment by mkasemer** [comment, marked as answer]
@Ming-is: regarding knowing which elements belong to which grain, this is printed with the mesh file. See Neper's documentation for further description of what is included with the mesh.

**Reply by rquey** [reply]
See also `-statelt3d elset3d` to output the elset of each element and `-statelset3d elts` to output the elements of each elset...

**Reply by Ming-is** [reply]
The fepx‘s result orientation is output according to the element, and the coordinates are output according to the node. Their numbers are different. I want to correspond them. So I used the command  **- reselt coo** to output the element coordinates or **_-resnode ori_** to output the node orientation by S module of neper , but both failed. What can I do to solve this problem?

**Reply by mkasemer** [reply]
@Ming-is - yes, you are correct that orientations are elemental quantities, not nodal, and coordinates are nodal quantities, not elemental. Orientations are not mapped to nodal positions.

If you want to know which coordinates belong to which elements, you need the element connectivity which is printed in the mesh file.
